libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -109,10 +110,7 @@ bool ByteBuffer::advance(uint32_t n)
     return true;
 }
 
-/*
-  read len bytes without advancing the read pointer
- */
-uint32_t ByteBuffer::peekbytes(uint8_t *data, uint32_t len)
+int ByteBuffer::peekiovec(ByteBuffer::IoVec iovec[2], uint32_t len)
```

Comment:
Since you dont copy anything to iovec (what is intended) and just get pointer to it, from the time you "peek" iovec to the time you used it, the buffer may have changed and iovec is now invalid/with some garbage. That is specially a problem since this is a lookless class. And I think this can hurt peekbytes very bad.